src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/DependenciesGraphProvider.cs

```diff
@@ -58,37 +46,44 @@ internal class DependenciesGraphProvider : OnceInitializedOnceDisposedAsync, IGr
                 trackChanges: true)
         };
 
-        /// <summary>
-        /// All icons that are used tree graph, register their monikers once to avoid extra UI thread switches.
-        /// </summary>
-        private ImmutableHashSet<ImageMoniker> _knownIcons = ImmutableHashSet<ImageMoniker>.Empty;
-
-        [ImportMany]
-        private OrderPrecedenceImportCollection<IDependenciesGraphActionHandler> GraphActionHandlers { get; }
-
         private readonly object _snapshotChangeHandlerLock = new object();
-        private IVsImageService2 _imageService;
-        private readonly object _expandedGraphContextsLock = new object();
 
         /// <summary>
         /// Remembers expanded graph nodes to track changes in their children.
         /// </summary>
-        protected WeakCollection<IGraphContext> ExpandedGraphContexts { get; } = new WeakCollection<IGraphContext>();
+        private readonly WeakCollection<IGraphContext> _expandedGraphContexts = new WeakCollection<IGraphContext>();
 
-        private IAggregateDependenciesSnapshotProvider AggregateSnapshotProvider { get; }
+        [ImportMany] private readonly OrderPrecedenceImportCollection<IDependenciesGraphActionHandler> _graphActionHandlers;
 
-        private IAsyncServiceProvider ServiceProvider { get; }
+        private readonly IAggregateDependenciesSnapshotProvider _aggregateSnapshotProvider;
+
+        private readonly IAsyncServiceProvider _serviceProvider;
+
+        private GraphIconCache _iconCache;
+
+        [ImportingConstructor]
+        public DependenciesGraphProvider(
```

Comment:
Why move the constructor around?

Reply:
Short answer is: it makes more sense to me this way.

Longer answer is actually a question: Do we have a standard order for members? I tend to use:

- static fields
- static members
- instance fields
- constructors
- computed properties
- methods

Reply:
I don't have a preference for ordering.  My only line in the sand is I don't want to be unable to debug because something is not in the correct order.  So I am not super-motivated to enforce anything with analyzers at the moment.

My only objection is that moving things around makes the diff harder to review.  I would prefer if style changes were just separate PRs.

Reply:
Also not convinced we need analyzers for this.

Understood regarding style changes. Sometimes it's only after moving things around that I see patterns such as pulling out related fields into a new class, as done here. Each commit here is straightforward.